service/frontend/accessControlledHandler.go

```diff
@@ -42,7 +43,10 @@ type AccessControlledWorkflowHandler struct {
 var _ Handler = (*AccessControlledWorkflowHandler)(nil)
 
 // NewAccessControlledHandlerImpl creates frontend handler with authentication support
-func NewAccessControlledHandlerImpl(wfHandler Handler, resource resource.Resource, authorizer authorization.Authorizer) *AccessControlledWorkflowHandler {
+func NewAccessControlledHandlerImpl(wfHandler Handler, resource resource.Resource, authorizer authorization.Authorizer, cfg config.Authorization) *AccessControlledWorkflowHandler {
+	if authorizer == nil {
```

Comment:
In the old code this is called by [this file](https://github.com/uber/cadence/blob/master/service/frontend/service.go#L258) that is checking already if the Authorizer is nil. This is being built [here](https://github.com/uber/cadence/blob/master/service/frontend/service.go#L182) and in both places where `NewService` is being called [here](https://github.com/uber/cadence/blob/master/host/onebox.go#L428) and [here](https://github.com/uber/cadence/blob/master/cmd/server/cadence/server.go#L232) the Authorizer is being populated. I'm wondering if we need this validation at all (and I was the one whoe removed it last time, sorry if it was needed)

Reply:
This will also be called by Uber’s internal repo which may not have authorizer. It’s just to be compatible